### PR TITLE
Deduplicate resource definition code

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:

Dedup top code duplications noted by sonarcloud.

[Overall Duplications](https://sonarcloud.io/component_measures?id=kubevirt_containerized-data-importer&metric=duplicated_lines_density&view=list) before this PR :
Density 18.9%
Duplicated Lines 6,529
Duplicated Blocks 113
Duplicated Files 27

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/operator/resources/cluster/cdiconfig.go [hunk 1]
```diff
@@ -36,497 +36,263 @@ func createCDIConfigCRD() *extv1.CustomResourceDefinition {
 					Name:    "v1alpha1",
 					Served:  true,
 					Storage: false,
-					Schema: &extv1.CustomResourceValidation{
-						OpenAPIV3Schema: &extv1.JSONSchemaProps{
-							Description: "CDIConfig provides a user configuration for CDI",
+					Schema:  createCDIConfigCRDSchema(),
```
Comment (awels) [inline pkg/operator/resources/cluster/cdiconfig.go:39]: Right here we assume that alphav1 and betav1 always remain the same, which is currently the case, but once we convince KubeVirt to use betav1, we won't update alphav1 anymore. So we would need to diverge the two at that point. Maybe start that by having a createAlphaV1ConfigCRDSchema and createBetaV1ConfigCRDSchema, and for now have alpha call beta, and put in a note that they might start diverging.

**Review by awels — COMMENTED** [review]
The same comment is true for all 3 files. The function assumes that both alphav1 and betav1 are the same and this is currently the case, but hopefully soon won't be the case.

File: pkg/operator/resources/cluster/cdiconfig.go (hunk 1 above)
Comment (arnongilboa) [reply pkg/operator/resources/cluster/cdiconfig.go]: @awels done.

**Comment by brybacki** [comment]
This is just an opinion: "Deduplicate resource definition code"

I know that the files changed could be easily seen, but having "resources" mentioned in the commit log might help if you plan to continue to deduplicate in some more places. But maybe you think having a series of the same comment is better.
like:
-  "Dedup top code duplications noted by sonarcloud"
-  "Dedup top code duplications noted by sonarcloud"

**Comment by arnongilboa** [comment]
> This is just an opinion: "Deduplicate resource definition code"

@brybacki , much better title, thanks!

**Pull request merged**